pkg/util/profiling/profiling.go

```diff
@@ -53,6 +52,12 @@ func Start(settings Settings) error {
 		profiler.WithAPIKey(""), // to silence the error log about `DD_API_KEY`
 	}
 
+	if settings.Socket != "" {
```

Comment:
I don't think its correct to make these mutually exclusive.

[`WithUDS`](https://github.com/DataDog/dd-trace-go/blob/main/profiler/options.go#L495) causes traces to be sent to the specified socket.

[`WithURL`](https://github.com/DataDog/dd-trace-go/blob/main/profiler/options.go#L355) simply sets the `apiURL` to the specified url, but does _not_ cause data to be sent there without `agentless` also being specified.

See [this line](https://github.com/DataDog/dd-trace-go/blob/main/profiler/profiler.go#L148) and the surrounding code.

I think the `WithURL` is quite confusing by itself, so I'd welcome a better comment documenting this behavior, or a change to add in support for specifying [`WithAgentlessUpload`](https://github.com/DataDog/dd-trace-go/blob/main/profiler/options.go#L335) and then have this mutually-exclusive behavior.

Reply:
Just noting that this is my understanding from reading the dd-trace-go code, but it could be worth having someone from @DataDog/tracing-go give their 2c

Reply:
I agree with your understanding here. It doesn't look like `internal_profiling.profile_dd_url` can do anything without additional configs. That lines up with my observations from last week.

I updated the comments and removed the mutual exclusion, but I didn't modify the existing behavior.